crates/client/l1-messages/src/contract.rs

```diff
@@ -0,0 +1,53 @@
+use ethers::contract::abigen;
+use ethers::types::U256;
+use mp_felt::Felt252Wrapper;
+use mp_transactions::HandleL1MessageTransaction;
+use starknet_api::transaction::Fee;
+
+use crate::error::L1MessagesWorkerError;
+
+abigen!(
+    L1Contract,
+    r"[
+	event LogMessageToL2(address indexed fromAddress, uint256 indexed toAddress, uint256 indexed selector, uint256[] payload, uint256 nonce, uint256 fee)
+]"
+);
+
+impl TryFrom<&LogMessageToL2Filter> for Fee {
+    type Error = L1MessagesWorkerError;
+
+    fn try_from(event: &LogMessageToL2Filter) -> Result<Self, Self::Error> {
+        // Check against panic
+        // https://docs.rs/ethers/latest/ethers/types/struct.U256.html#method.as_u128
+        if event.fee > U256::from_big_endian(&(u128::MAX.to_be_bytes())) {
+            Err(L1MessagesWorkerError::ToFeeError)
+        } else {
+            Ok(Fee(event.fee.as_u128()))
+        }
+    }
+}
```

Comment:
This look more like a getter than a real Into.

I would either:
- impl it as a getter on LogMessageToL2Filter
or
- imp `TryFrom` once: `TryFrom<LogMessageToL2Filter> for (Fee, HandleL1MessageTransaction)`

You call both one after the other anyway.

Reply:
Ok